compiler+runtime/src/cpp/jank/util/dir.cpp

```diff
@@ -61,15 +62,20 @@ namespace jank::util
     return res;
   }
 
-  native_persistent_string const &binary_cache_dir()
+  native_persistent_string const &
+  binary_cache_dir(native_integer const optimization_level,
+                   native_vector<native_persistent_string> const &includes,
+                   native_vector<native_persistent_string> const &defines)
   {
     static native_persistent_string res;
     if(!res.empty())
     {
       return res;
     }
 
-    return res = fmt::format("{}/{}", user_cache_dir(), binary_version());
+    return res = fmt::format("{}/{}",
+                             user_cache_dir(),
+                             binary_version(optimization_level, includes, defines));
```

Comment:
Hm, I was wrong, Saket. This is not going to work. If two separate projects both have ns `a`, with different code, then compiling one project first, then going to compile the other project could result in loading the first project's `a` when you go to compile the second.

This could work fine for dependencies from maven, but we don't have a good way to distinguish them from project sources right now. To mitigate this, let's put the binary cache dir in `target/<binary version>`, relative to the current directory. This is what leiningen does. So we should just be able to replace `user_cache_dir` here with `"target"`.

Reply:
Uhuh. That is a good observation. Thanks for catching.

Dependency and project modules could also conflict. But I think it can be the developer's responsibility to choose unique prefixes for their namespaces.

Not as important to handle this within language than other features, at the moment.